org.eclipse.buildship.kotlin/src/main/kotlin/org/eclipse/buildship/kotlin/KotlinBuildScript.kt

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2017 the original author or authors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.eclipse.buildship.kotlin
+
+import org.gradle.api.Project
+import org.gradle.api.plugins.ObjectConfigurationAction
+import org.gradle.script.lang.kotlin.KotlinScriptHandler
+import org.jetbrains.kotlin.script.ScriptTemplateDefinition
+
+@ScriptTemplateDefinition(
+    resolver = GradleKotlinScriptDependenciesResolver::class,
+    scriptFilePattern = ".*\\.gradle\\.kts")
+abstract class KotlinBuildScript(project: Project) : Project by project {
```

Comment:
Here the `resolver = GradleKotlinScriptDependenciesResolver::class,` part is the interesting one: it references the class from the `org.eclipse.buildship.kotlin` plugin instead of from GSK.